spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/annotation/SleuthAdvisorConfig.java

```diff
@@ -183,44 +183,138 @@ public Object invoke(MethodInvocation invocation) throws Throwable {
 		if (method == null) {
 			return invocation.proceed();
 		}
+
 		Method mostSpecificMethod = AopUtils
 				.getMostSpecificMethod(method, invocation.getThis().getClass());
 		NewSpan newSpan = SleuthAnnotationUtils.findAnnotation(mostSpecificMethod, NewSpan.class);
 		ContinueSpan continueSpan = SleuthAnnotationUtils.findAnnotation(mostSpecificMethod, ContinueSpan.class);
 		if (newSpan == null && continueSpan == null) {
 			return invocation.proceed();
 		}
+
+		if(isReactiveReturnType(method.getReturnType())){
+			return proceedUnderSpanReactively(invocation, newSpan, continueSpan);
+		} else {
+			return proceedUnderSpanSynchronously(invocation, newSpan, continueSpan);
+		}
+	}
+
+	private boolean isReactiveReturnType(Class<?> returnType) {
+		return Flux.class.equals(returnType) || Mono.class.equals(returnType);
```

Comment:
Write a javadoc or comment about how we know this is the universe of return types. For example, This implies there are no subtypes of Flux or Mono, and also that there are no other things such as Flowable that can be returned.

Reply:
renamed so now reactive scope is clear from the name